### PR TITLE
Fix: prefer-const autofix (fixes #10582)

### LINKED ISSUE #10582: prefer-const should autofix, but it didn't
```js
let acorn = require("acorn"),
    assert = require("chai").assert;
```
to
```js
const acorn = require("acorn"),
    assert = require("chai").assert;
```

### PR DESCRIPTION
**Pull request by sstern6**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Removed the guard checking for multiline var declarations to ensure autofix is applied.

Issue: https://github.com/eslint/eslint/issues/10582

**Is there anything you'd like reviewers to focus on?**
N/A

### CODE REVIEW

File: tests/lib/rules/prefer-const.js [hunk 1]
```diff
@@ -309,7 +309,7 @@ ruleTester.run("prefer-const", rule, {
         },
         {
             code: "let {a = 0, b} = obj, c = a; b = a;",
-            output: null,
+            output: "const {a = 0, b} = obj, c = a; b = a;",
```
Comment (platinumazure) [inline tests/lib/rules/prefer-const.js]: It looks like this result will have a runtime error as `b` is reassigned after being declared `const`. Are you sure this is the right approach?

File: tests/lib/rules/prefer-const.js [hunk 2]
```diff
@@ -318,7 +318,7 @@ ruleTester.run("prefer-const", rule, {
         },
         {
             code: "let {a = 0, b} = obj, c = a; b = a;",
-            output: null,
+            output: "const {a = 0, b} = obj, c = a; b = a;" ,
```
Comment (platinumazure) [inline tests/lib/rules/prefer-const.js]: Previous comment applies here as well.

**Comment by platinumazure** [comment]
> The rule should guard against creating runtime errors like above?

Yes- autofix should never result in incorrect code.

**Comment by sstern6** [comment]
Thanks @platinumazure will rework this PR, never worked with the auto-fixing aspect of a rule and this completely slipped my mind.

File: tests/lib/rules/prefer-const.js (hunk 1 above)
Comment (sstern6) [reply tests/lib/rules/prefer-const.js]: Sounds good @kaicataldo

**Comment by sstern6** [comment]
@platinumazure @kaicataldo PR and tests back up, let me know if you have any questions or comments.

Thanks

File: lib/rules/prefer-const.js [hunk 3]
```diff
@@ -357,6 +357,9 @@ module.exports = {
         const shouldMatchAnyDestructuredVariable = options.destructuring !== "all";
         const ignoreReadBeforeAssign = options.ignoreReadBeforeAssign === true;
         const variables = [];
+        let reportCount = 0;
+        const firstIdentifier = sourceCode.ast.tokens.find(node => node.type === "Identifier");
```
Comment (platinumazure) [inline lib/rules/prefer-const.js]: I'm confused about this line. It looks like this line will grab the first identifier token in the program, without consideration for scope.

If you add a few tests that have some nested scopes, you might see some issues. For example:

```js
const foo = 1;   // First identifier token is on this line

function someFunc() {
    let a = 0,
        b = 1;

    bar(a, b);
    // Both a and b should be made const in this case, but are they detected properly?
}
```

Comment (sstern6) [reply lib/rules/prefer-const.js]: @platinumazure thanks for the feedback, good point, didnt think about this but yes this is detected properly.  The line you commented on is just an initializer and then compares current with previous name.  I will add tests for the various scoping options as well.

**Comment by sstern6** [comment]
@platinumazure just following up on this, let me know if you have any questions or comments that I can address.  Thank you!

**Comment by sstern6** [comment]
Bump.

**Comment by platinumazure** [comment]
The new test cases don't seem to be multiline-- I think the focus of this issue has changed if I'm not mistaken? I think the goal now is to autofix when we know all parts of the declaration (all declarators in a VariableDeclaration, all properties in an ObjectPattern) can be made `const`, and whether something is multiline or not doesn't matter. Let me know if I'm mistaken... Thanks for your patience!

**Comment by sstern6** [comment]
@platinumazure youre right, it shouldnt matter if its multiline or not, Ill reformat the tests to be multiline.

Sound good?

For example, the code found https://github.com/eslint/eslint/pull/10987/files#diff-7bdeecf4f450247ac04335fce20a1e85R459.

I can change the code from `let x = 'x', y = 'y';`

to: 
```
let x = 'x',
  y = 'y';
```

How does this sound?  Did I answer your question?

**Comment by platinumazure** [comment]
As we discussed in Gitter-- the tests are fine, it's just that the commit message/PR title aren't quite accurate anymore. If others on the team are okay with this, we could just handle it as we merge.

I'm going to leave this open another couple of days for others to review, just in case.

**Comment by nzakas** [comment]
Thanks @sstern6!

**Pull request merged**